src/app/common/directives/header-anchor.directive.ts

```diff
@@ -0,0 +1,36 @@
+import {AfterViewInit, Directive, ElementRef, Input, Renderer2} from '@angular/core';
+import {ActivatedRoute, Router} from '@angular/router';
+
+@Directive({
+  selector: 'h3,h4',
+})
+export class HeaderAnchorDirective implements AfterViewInit {
+
+  @Input('id') public overriddenId: string | undefined;
+
+  constructor (private elementRef: ElementRef<HTMLHeadingElement>,
+               private renderer: Renderer2,
+               private route: ActivatedRoute,
+               private router: Router) {
+  }
+
+  private getImplicitId (): string {
+    const content = this.elementRef.nativeElement.innerText;
+    return content.replace(/\s/g, '-').toLocaleLowerCase();
+  }
+
+  public ngAfterViewInit (): void {
+    const id = this.overriddenId == null ? this.getImplicitId() : this.overriddenId;
```

Comment:
If the id is not yet set, I would update `this.overridenId` instead of creating another local variable id here.
(Especially, if you rename `overridenId` to `id`.)